src/program.ts

```diff
@@ -192,118 +192,180 @@ export enum OperatorKind {
   // LOGICAL_OR           // a || b
 }
 
-/** Returns the operator kind represented by the specified decorator and string argument. */
-function operatorKindFromDecorator(decoratorKind: DecoratorKind, arg: string): OperatorKind {
-  assert(arg.length);
-  switch (decoratorKind) {
-    case DecoratorKind.OPERATOR:
-    case DecoratorKind.OPERATOR_BINARY: {
-      switch (arg.charCodeAt(0)) {
-        case CharCode.OPENBRACKET: {
-          if (arg == "[]") return OperatorKind.INDEXED_GET;
-          if (arg == "[]=") return OperatorKind.INDEXED_SET;
-          break;
-        }
-        case CharCode.OPENBRACE: {
-          if (arg == "{}") return OperatorKind.UNCHECKED_INDEXED_GET;
-          if (arg == "{}=") return OperatorKind.UNCHECKED_INDEXED_SET;
-          break;
-        }
-        case CharCode.PLUS: {
-          if (arg == "+") return OperatorKind.ADD;
-          break;
-        }
-        case CharCode.MINUS: {
-          if (arg == "-") return OperatorKind.SUB;
-          break;
-        }
-        case CharCode.ASTERISK: {
-          if (arg == "*") return OperatorKind.MUL;
-          if (arg == "**") return OperatorKind.POW;
-          break;
-        }
-        case CharCode.SLASH: {
-          if (arg == "/") return OperatorKind.DIV;
-          break;
-        }
-        case CharCode.PERCENT: {
-          if (arg == "%") return OperatorKind.REM;
-          break;
-        }
-        case CharCode.AMPERSAND: {
-          if (arg == "&") return OperatorKind.BITWISE_AND;
-          break;
-        }
-        case CharCode.BAR: {
-          if (arg == "|") return OperatorKind.BITWISE_OR;
-          break;
-        }
-        case CharCode.CARET: {
-          if (arg == "^") return OperatorKind.BITWISE_XOR;
-          break;
-        }
-        case CharCode.EQUALS: {
-          if (arg == "==") return OperatorKind.EQ;
-          break;
-        }
-        case CharCode.EXCLAMATION: {
-          if (arg == "!=") return OperatorKind.NE;
-          break;
-        }
-        case CharCode.GREATERTHAN: {
-          if (arg == ">") return OperatorKind.GT;
-          if (arg == ">=") return OperatorKind.GE;
-          if (arg == ">>") return OperatorKind.BITWISE_SHR;
-          if (arg == ">>>") return OperatorKind.BITWISE_SHR_U;
-          break;
-        }
-        case CharCode.LESSTHAN: {
-          if (arg == "<") return OperatorKind.LT;
-          if (arg == "<=") return OperatorKind.LE;
-          if (arg == "<<") return OperatorKind.BITWISE_SHL;
-          break;
+export namespace OperatorKind {
+
+  /** Returns the operator kind represented by the specified decorator and string argument. */
+  export function fromDecorator(decoratorKind: DecoratorKind, arg: string): OperatorKind {
+    assert(arg.length);
+    switch (decoratorKind) {
+      case DecoratorKind.OPERATOR:
+      case DecoratorKind.OPERATOR_BINARY: {
+        switch (arg.charCodeAt(0)) {
```

Comment:
It seems `arg.charCodeAt(0)` could be moved on top and cached with re-usage result here and in L#272, L#295

Reply:
Difference is 3 calls vs 1 call + 1 local.set + 3 local.get, which seems about the same size-wise. The function can also be used to determine whether something is an operator overload, in which case, if it isn't one, the first char isn't needed. I guess it's fine :)